nifi-nar-bundles/nifi-avro-bundle/nifi-avro-processors/src/main/java/org/apache/nifi/processors/avro/SplitAvro.java

```diff
@@ -267,7 +267,8 @@ public void process(InputStream rawIn) throws IOException {
                         }
 
                         // while records are left, start a new split by spawning a FlowFile
-                        while (reader.hasNext()) {
+                        final AtomicReference<Boolean> hasNextHolder = new AtomicReference<Boolean>(reader.hasNext());
```

Comment:
If there are no records, then the call to reader.next() [will throw a NoSuchElementException](https://avro.apache.org/docs/1.7.7/api/java/org/apache/avro/file/DataFileStream.html#next(D)). You could wrap the code in a try/catch, but then it's a little heavier, so the AtomicReference is probably fine?